File: Leetcode/2001-3000/2601-2700/2697. Lexicographically Smallest Palindrome.py
class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        l = list(s)
        left, right = 0, len(s) - 1
        while left < right:
            if l[left] != l[right]:
                l[left] = l[right] = l[left] if ord(l[left]) < ord(l[right]) else l[right]
            left += 1
            right -= 1
        return ''.join(l)

File: Leetcode/2001-3000/2601-2700/2697. Lexicographically Smallest Palindrome.go
// https://leetcode.com/problems/lexicographically-smallest-palindrome/

package leetcode

import (
	"strings"
)

func makeSmallestPalindrome(s string) string {
	temp := strings.Split(s, "")
	l := len(s)
	for i := 0; i < l/2; i++ {
		if temp[i] != temp[l-i-1] {
			if temp[i] < temp[l-i-1] {
				temp[l-i-1] = temp[i]
			} else {
				temp[i] = temp[l-i-1]
			}
		}
	}
	return strings.Join(temp, "")
}
